scripts/case_selection/export_bpc_selected_cases.R

```diff
@@ -120,35 +120,44 @@ existing_patients = selected_cases[selected_cases %in% clinical$patient_id]
-
-patient_output$genie_patient_id <- patient_output$record_id
-patient_output$birth_year <- clinical$birth_year[match(patient_output$genie_patient_id, clinical$patient_id)]
-patient_output$naaccr_ethnicity_code <- clinical$ethnicity[match(patient_output$genie_patient_id, clinical$patient_id)]
-patient_output$naaccr_race_code_primary <- clinical$primary_race[match(patient_output$genie_patient_id, clinical$patient_id)]
-patient_output$naaccr_race_code_secondary <- clinical$secondary_race[match(patient_output$genie_patient_id, clinical$patient_id)]
-patient_output$naaccr_race_code_tertiary <- clinical$tertiary_race[match(patient_output$genie_patient_id, clinical$patient_id)]
-patient_output$naaccr_sex_code <- clinical$sex[match(patient_output$genie_patient_id, clinical$patient_id)]
-
-# mapping to code
-patient_output$naaccr_ethnicity_code <- ethnicity_mapping$CODE[match(patient_output$naaccr_ethnicity_code, ethnicity_mapping$CBIO_LABEL)]
-patient_output$naaccr_race_code_primary <- race_mapping$CODE[match(patient_output$naaccr_race_code_primary, race_mapping$CBIO_LABEL)]
-patient_output$naaccr_race_code_secondary <- race_mapping$CODE[match(patient_output$naaccr_race_code_secondary, race_mapping$CBIO_LABEL)]
-patient_output$naaccr_race_code_tertiary <- race_mapping$CODE[match(patient_output$naaccr_race_code_tertiary, race_mapping$CBIO_LABEL)]
-patient_output$naaccr_sex_code <- sex_mapping$CODE[match(patient_output$naaccr_sex_code,sex_mapping$CBIO_LABEL)]
+#'  mapping data for instrument - patient_characteristics
+#' 
+#' @param clinical A data frame of released clinical data for selected cases
+#' @param existing_patients A data frame of available patient after case selection
+#' @return A data frame with mapped code
+remap_patient_characteristics <- function(clinical, existing_patients, ethnicity_mapping, race_mapping, sex_mapping){
+  
+  patient_df <- data.frame("record_id" = existing_patients)
+  patient_df$redcap_repeat_instrument <- rep("")
+  patient_df$redcap_repeat_instance <- rep("")
+  
+  patient_df$genie_patient_id <- patient_df$record_id
+  patient_df$birth_year <- clinical$birth_year[match(patient_df$genie_patient_id, clinical$patient_id)]
+  patient_df$naaccr_ethnicity_code <- clinical$ethnicity_detailed[match(patient_df$genie_patient_id, clinical$patient_id)]
+  patient_df$naaccr_race_code_primary <- clinical$primary_race_detailed[match(patient_df$genie_patient_id, clinical$patient_id)]
+  patient_df$naaccr_race_code_secondary <- clinical$secondary_race_detailed[match(patient_df$genie_patient_id, clinical$patient_id)]
+  patient_df$naaccr_race_code_tertiary <- clinical$tertiary_race_detailed[match(patient_df$genie_patient_id, clinical$patient_id)]
+  patient_df$naaccr_sex_code <- clinical$sex_detailed[match(patient_df$genie_patient_id, clinical$patient_id)]
+  
+  # mapping to code
+  patient_df$naaccr_ethnicity_code <- ethnicity_mapping$CODE[match(patient_df$naaccr_ethnicity_code, ethnicity_mapping$DESCRIPTION)]
```

Comment:
Do we want to throw an error if there isn't a match?

Reply:
I can add check here.

Reply:
I had the second thought. This might not be a issue due to line `existing_patients = selected_cases[selected_cases %in% clinical$patient_id]`. Let me test it out.

Reply:
Hmmm, I am concerned about strings not mapping to DESCRIPTION.

Reply:
That shouldn't be an issue since the detailed columns are actually generated from DESCRIPTION in the table.

Reply:
I see your point. Ya, This could be an issue if we update DESCRIPTION column in the mapping tables, but the clinical file is still using the old values in the detailed columns. If this it the case, then the code would render NAs for the row with mismatch detailed column and DESCRIPTION column. Do we want to throw warning for this?

Reply:
Yes, i think it makes sense to check for NAs in the final output just in case there are any.  So if there is no match, it becomes

```
ethnicity_mapping$CODE[NA]
```

Does that cause an error?

Reply:
I won't cause an error in this code but I'm not sure if this could break downstream processing. @Chelsea-Na Do you know if we allow NAs in naaccr code columns?

Reply:
So back when I was testing staging workflow for main Genie, I noted that the center JHU had nans in `SECONDARY_RACE`, `TERTIARY_RACE` columns in the production project's `clinical.txt` but Not Collected in those same columns in staging. So there were definitely NAs it seems. 

Even though we have validation [here](https://github.com/Sage-Bionetworks/Genie/blob/develop/genie_registry/clinical.py#L1060-L1106) that should not be letting NAs through in those columns

Reply:
Currently, the code will break if NAs or empty strings are found in naaccr code columns.

Reply:
That's going to be a problem because there are existing nans in the `data_clinical.txt`

Reply:
@danlu1 / @rxu17 can we bring the discussion in #genie slack on what we want to do with null secondary and tertiary race values.  The sites impacted are  CHOP, JHU, and PROV.

Reply:
Yes, we can easily convert to throw warning instead of error. "" are in the naaccr code columns. 

<img width="1233" alt="Screenshot 2024-09-13 at 11 08 53 AM" src="https://github.com/user-attachments/assets/40d31088-e866-4dd7-a61f-7b10ac09c02a">